fix: Rename Doc considers invalid table name on renaming record

marination opened the pull request:
**Issue:**
- Consider a Doctype "Salary Component", which has a **record** called **Additionsl Salary**(typo)
- There is a pre-existing Doctype called **Additional Salary**
- Rename **record** **Additionsl Salary** to **Additional Salary**:
  ```
  File "/Users/marica/Desktop/bench-v-12/apps/frappe/frappe/model/rename_doc.py", line 259, in update_link_field_values
    fieldname=field['fieldname']), (new, old))
  File "/Users/marica/Desktop/bench-v-12/apps/frappe/frappe/database/database.py", line 147, in sql
    self._cursor.execute(query, values)
  File "/Users/marica/Desktop/bench-v-12/env/lib/python3.7/site-packages/pymysql/cursors.py", line 170, in execute
    result = self._query(query)
  File "/Users/marica/Desktop/bench-v-12/env/lib/python3.7/site-packages/pymysql/cursors.py", line 328, in _query
    conn.query(q)
  File "/Users/marica/Desktop/bench-v-12/env/lib/python3.7/site-packages/pymysql/connections.py", line 517, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File "/Users/marica/Desktop/bench-v-12/env/lib/python3.7/site-packages/pymysql/connections.py", line 732, in _read_query_result
    result.read()
  File "/Users/marica/Desktop/bench-v-12/env/lib/python3.7/site-packages/pymysql/connections.py", line 1075, in read
    first_packet = self.connection._read_packet()
  File "/Users/marica/Desktop/bench-v-12/env/lib/python3.7/site-packages/pymysql/connections.py", line 684, in _read_packet
    packet.check_error()
  File "/Users/marica/Desktop/bench-v-12/env/lib/python3.7/site-packages/pymysql/protocol.py", line 220, in check_error
    err.raise_mysql_exception(self._data)
  File "/Users/marica/Desktop/bench-v-12/env/lib/python3.7/site-packages/pymysql/err.py", line 109, in raise_mysql_exception
    raise errorclass(errno, errval)
  pymysql.err.ProgrammingError: (1146, "Table '_35f9c674cad210fe.tabadditionsl salary' doesn't exist")
   ```
- This happened as the new record value coincides with a related Link field's parent name or DocType name

**Fix:**
- **By default** the parent field / table to update, should have the related Link field's parent name or DocType name, as, a field within this doctype is to be updated. While renaming records, the comparison between the old/new values and `field["parent"]` does not make sense.
- Parent value should only be considered for any conditional change if a DocType is being renamed
- Use case for the conditional bit is: Having "DocType" link field (DocType as the Options) in the DocType you are trying to rename

**Other:**
- Added Postgres support for `frappe.rename_doc` queries
- Removed dead code in `rename_doc.py`

marination committed:
fix: Conditionally set parent field only on DocType rename

marination committed:
Merge branch 'develop' of https://github.com/frappe/frappe into rename-doc-rename-link-fields

gavindsouza commented on frappe/model/rename_doc.py:
```diff
@@ -249,8 +249,11 @@ def update_link_field_values(link_fields, old, new, doctype):
+
 			# because the table hasn't been renamed yet!
-			parent = field['parent'] if field['parent']!=new else old
+			if field['parent'] == new and doctype == "DocType":
```
```suggestion
			if parent == new and doctype == "DocType":
```

Add a comment explaining why we need to do this here, too.

marination committed:
chore: Rename Doctype Test and more explicit comment

- Better decription of why the fix is done, what case it handles
- Test for Renaming Doctype and Record having same name as DocType

marination commented:
@Mergifyio backport version-12-hotfix

gavindsouza committed:
Merge branch 'develop' into rename-doc-rename-link-fields

gavindsouza committed:
fix: Add postgreSQL support for rename_doc queries

gavindsouza committed:
chore: Remove dead code

gavindsouza committed:
Merge branch 'develop' of github.com:frappe/frappe into rename-doc-rename-link-fields
